torch/csrc/autograd/python_torch_functions_manual.cpp

```diff
@@ -484,6 +484,26 @@ static PyObject* THPVariable__enable_functionalization(
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject* THPVariable__functionalize_enable_reapply_views(
+    PyObject* self,
+    PyObject* args,
+    PyObject* kwargs) {
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser(
+      {"_functionalize_enable_reapply_views(bool reapply_views=False)"},
+      /*traceable=*/true);
+  ParsedArgs<1> parsed_args;
+  auto r = parser.parse(args, kwargs, parsed_args);
+  const auto reapply_views = r.toBool(0);
+  if (reapply_views) {
+    at::functionalization::impl::setFunctionalizationReapplyViewsTLS(true);
+  } else {
+    at::functionalization::impl::setFunctionalizationReapplyViewsTLS(false);
```

Comment:
Don't you already have a binding THPVariable__enable_functionalization for this

Reply:
I explicitly added this API because `THPVariable__enable_functionalization` does two things:

(1) sets `reapply_views` TLS to true/false

(2) puts the `Functionalize` key into the dispatcher TLS

And in my usage of functional tensor above, I'm using the fancy new context manager `_IncludeDispatchKeyGuard` to do (1). So I figured I'd separate out these API's.

Some alternatives:

(a) I could also turn this into a python context manager

(b) I could have a dedicated context manager for both (1) and (2), since they're usually used together

I'll probably do one of the two options above, but happy to do either